### PR TITLE
hybpiper: new package

### PR DESCRIPTION
**Pull request by akthoma**
Adding new package hybpiper along with spades dependency

### CODE REVIEW

File: var/spack/repos/builtin/packages/hybpiper/package.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+       especially for phylogenetics. HybPiper is a suite of Python scripts
+       that wrap and connect bioinformatics tools in order to extract target
+       sequences from high-throughput DNA sequencing reads"""
+
+    homepage = "https://github.com/mossmatters/HybPiper"
+    url      = "https://github.com/mossmatters/HybPiper/archive/v1.2.0.tar.gz"
+
+    version('1.2.0', '0ad78e9ca5e3f23ae0eb6236b07e1780')
+
+    depends_on('python@2.7:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hybpiper/package.py]: Is Python needed at build time, at run time, or for linking?

File: var/spack/repos/builtin/packages/spades/package.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+class Spades(Package):
+    """SPAdes - St. Petersburg genome assembler - is intended for both
+       standard isolates and single-cell MDA bacteria assemblies."""
+
+    homepage = "http://www.example.com"
+    url      = "http://cab.spbu.ru/files/release3.10.1/SPAdes-3.10.1.tar.gz"
+
+    version('3.10.1', 'dcab7d145af81b59cc867562f27536c3')
+
+    depends_on('python')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/spades/package.py]: What dependency type should this be?

File: var/spack/repos/builtin/packages/spades/package.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    depends_on('cmake', type='build')
+    depends_on('zlib')
+    depends_on('bzip2')
+
+    conflicts('%gcc@7.1.0:')
+
+    def install(self, spec, prefix):
+        os.environ['PREFIX'] = prefix
+        subprocess.call(['./spades_compile.sh'])
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/spades/package.py]: `subprocess.call` was added in Python 2.7. It won't work in 2.6.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/spades/package.py]: I took a look at this script and it just runs CMake like any other `CMakePackage` does, but without any of the flags that Spack adds by default. I think the best course of action would be to convert this package to `CMakePackage` and remove the `install()` function altogether.

File: var/spack/repos/builtin/packages/hybpiper/package.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+       that wrap and connect bioinformatics tools in order to extract target
+       sequences from high-throughput DNA sequencing reads"""
+
+    homepage = "https://github.com/mossmatters/HybPiper"
+    url      = "https://github.com/mossmatters/HybPiper/archive/v1.2.0.tar.gz"
+
+    version('1.2.0', '0ad78e9ca5e3f23ae0eb6236b07e1780')
+
+    # Only needed for building docs
+    depends_on('python@2.7:', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hybpiper/package.py]: You say it's only needed for docs, but is it not also needed for `py-biopython`?

Comment (akthoma) [inline var/spack/repos/builtin/packages/hybpiper/package.py]: My apologies, I'm still trying to get used to the python module setup. My thinking here was that `py-biopython` would have it's own dependency for python in it's spec, and the only time `hybpiper` exclusively used python was for building docs.

File: var/spack/repos/builtin/packages/spades/package.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    """SPAdes - St. Petersburg genome assembler - is intended for both
+       standard isolates and single-cell MDA bacteria assemblies."""
+
+    homepage = "http://cab.spbu.ru/software/spades/"
+    url      = "http://cab.spbu.ru/files/release3.10.1/SPAdes-3.10.1.tar.gz"
+
+    version('3.10.1', 'dcab7d145af81b59cc867562f27536c3')
+
+    depends_on('python', type=('build', 'run'))
+    depends_on('cmake', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/spades/package.py]: You don't need this line. It's already present in CMakePackage.

**Pull request merged**